pkg/metrics/metrics.go

```diff
@@ -565,6 +568,80 @@ var (
 	// APILimiterProcessedRequests is the counter of the number of
 	// processed (successful and failed) requests
 	APILimiterProcessedRequests = NoOpCounterVec
+
+	// WorkQueueDepth is the depth of the workqueue
+	//
+	// We set actual metrics here instead of NoOp for the workqueue metrics
+	// because these metrics will be registered with workqueue.SetProvider
+	// by init function in watcher.go. Otherwise, we will register NoOps.
+	//
+	WorkQueueDepth = metric.NewGaugeVec(metric.GaugeOpts{
+		ConfigName: Namespace + "_" + SubsystemWorkQueue + "_depth",
+		Namespace:  Namespace,
+		Subsystem:  SubsystemWorkQueue,
+		Name:       "depth",
+		Help:       "Current depth of workqueue",
+	}, []string{"name"})
+
+	// WorkQueueAddsTotal is the total number of adds to the workqueue
+	WorkQueueAddsTotal = metric.NewCounterVec(metric.CounterOpts{
+		ConfigName: Namespace + "_" + SubsystemWorkQueue + "_adds_total",
+		Namespace:  Namespace,
+		Subsystem:  SubsystemWorkQueue,
+		Name:       "adds_total",
+		Help:       "Total number of adds handled by workqueue",
+	}, []string{"name"})
+
+	// WorkQueueLatency is the latency of how long an item stays in the workqueue
+	WorkQueueLatency = metric.NewHistogramVec(metric.HistogramOpts{
+		ConfigName: Namespace + "_" + SubsystemWorkQueue + "_queue_duration_seconds",
+		Namespace:  Namespace,
+		Subsystem:  SubsystemWorkQueue,
+		Name:       "queue_duration_seconds",
+		Help:       "How long in seconds an item stays in workqueue before being requested",
```

Comment:
nit: requested vs processed?

Reply:
WorkQueueLatency is the latency of how long an item stays in workqueue before a consumer requests the item from it. The consumer will start processing the item after that.
https://github.com/kubernetes/client-go/blob/21b4553e4404bae3fec9ff9ef7204beabcfd3c27/util/workqueue/metrics.go#L112